Define BUILD_NUM_JOBS in TFLite Cmake script to prevent crash

feranick opened the pull request:
Reference: https://github.com/tensorflow/tensorflow/issues/63018

cc: @mihaimaruseac

feranick committed:
Define BUILD_NUM_JOBS in TFLite Cmake script to prevent crash

Reference: https://github.com/tensorflow/tensorflow/issues/63018

mihaimaruseac commented on tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh:
```diff
@@ -25,6 +25,7 @@ export PACKAGE_VERSION="${TENSORFLOW_VERSION}${VERSION_SUFFIX}"
 export PROJECT_NAME=${WHEEL_PROJECT_NAME:-tflite_runtime}
 BUILD_DIR="${SCRIPT_DIR}/gen/tflite_pip/${PYTHON}"
 TENSORFLOW_TARGET=${TENSORFLOW_TARGET:-$1}
+BUILD_NUM_JOBS=4
```
This forces it to 4, regardless if the environment sets a different value. We should only set it only if not already set (e.g., see line 19 for how `$PYTHON` is defined).

Also, for this one, can you add a comment?

feranick committed:
Define BUILD_NUM_JOBS value only if not set by environment

BUILD_NUM_JOBS is not set by default, so unless it is set by the environment, it leads to a boundless number of processes being started upon compilation. The previous patch set this value, but prevented override by the environement. This commit sets the value, but it allows this value to be overridden by the environment if the users explicitely decides to.
